lib/honeybadger/plugins/net_http.rb

```diff
@@ -0,0 +1,54 @@
+require "net/http"
+require 'honeybadger/plugin'
+require 'honeybadger/instrumentation'
+require "resolv"
+
+module Honeybadger
+  module Plugins
+    module Net
+      module HTTP
+        def request(request_data, body = nil, &block)
+          return super unless started?
+          return super if hb?
+
+          Honeybadger::Instrumentation.monotonic_timer { super }.tap do |duration, response_data|
+            context = {
+              duration: duration,
+              method: request_data.method,
+              status: response_data.code.to_i
+            }.merge(parsed_uri_data(request_data))
+
+            Honeybadger.event('request', context)
+          end[1] # return the response data only
+        end
+
+        def hb?
+          address.to_s[/#{Honeybadger.config[:'connection.host'].to_s}/]
+        end
+
+        def parsed_uri_data(request_data)
+          uri = request_data.uri || build_uri(request_data)
+
+          {
```

Comment:
I think it would be useful to always report the full url (rather than just when rescuing) so that url can always be queried. That is, I'd like to have the consistency of always having the same field available for queries. If the full url is always reported, I'm not sure there's much value in reporting the component parts, but perhaps you had a use-case in mind that I'm not considering.

Reply:
Right, this makes sense. I'm just going to send the full uri string.